src/App.tsx

```diff
@@ -1,24 +1,261 @@
+  const [selectedOption, setSelectedOption] = useState(TodoStatus.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [inputValue, setInputValue] = useState('');
+  const [isInputFieldDisabled, setIsInputFieldDisabled] = useState(false);
+
+  const updateTodo = (todo: Todo) => {
+    setLoadingTodosIds([todo.id]);
+
+    todosService.updateTodo(todo)
+      .then(updatedTodo => {
+        setLoadingTodosIds([]);
+        setTodos(prevState => prevState.map(currentTodo => (
+          currentTodo.id !== updatedTodo.id
+            ? currentTodo
+            : updatedTodo
+        )));
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+        setLoadingTodosIds([]);
+      });
+  };
+
+  const handleCompletedChange = (todoId: number) => {
+    const foundTodo = todos.find(todo => todo.id === todoId);
+
+    if (foundTodo) {
+      updateTodo({ ...foundTodo, completed: !foundTodo.completed });
+    }
+  };
+
+  useEffect(() => {
+    setIsLoadingTodos(true);
+
+    todosService
+      .getTodos()
+      .then((todosFromSrever) => {
+        setTodos(todosFromSrever);
+        setIsLoadingTodos(false);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setIsLoadingTodos(false);
+      });
+  }, []);
+
+  const visibleTodos = filterService.filterTodos(todos, selectedOption);
+
+  const handleChangeSelect = (newOption: TodoStatus) => {
+    setSelectedOption(newOption);
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoadingTodosIds([todoId]);
+
+    todosService.deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos => currentTodos
+          .filter(todo => todo.id !== todoId));
+        setLoadingTodosIds([]);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        setLoadingTodosIds([]);
+      })
+      .finally(() => setLoadingTodosIds([todoId]));
+  };
+
+  const handleDeleteAll = () => {
+    const filteredTodos = todos.filter((current) => current.completed);
+
+    setLoadingTodosIds(filteredTodos.map((current) => current.id));
+    Promise.allSettled(filteredTodos
+      .map(todo => todosService.deleteTodo(todo.id)))
+      .then((rezult) => {
+        const fulfilledTodoIds: number[] = [];
+        let wasFailed = false;
+
+        rezult.forEach((response, i) => {
+          if (response.status === 'fulfilled') {
+            fulfilledTodoIds.push(filteredTodos[i].id);
+          } else {
+            wasFailed = true;
+          }
+        });
+
+        if (wasFailed) {
+          setErrorMessage('Unable to delete a todo');
+        }
+
+        setLoadingTodosIds([]);
+        setTodos(currentTodos => currentTodos
+          .filter(current => !fulfilledTodoIds
+            .includes(current.id)));
+      });
+  };
+
+  const handleChangeAllCompleted = () => {
+    const isNewCompleted = todos.some(todo => todo.completed === false);
```

Comment:
What exactly is new completed?
```suggestion
    const isNewCompleted = todos.some(todo => !todo.completed);
```